### PR TITLE
Fixes for NetBSD

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
I'm always paranoid about touching any `lib/crypto/` stuff, so I separated all the `RSA -> void` changes into function-by-function for easier reviewing.  I can merge them all together if you want.

**Comment by gperciva** [comment]
Most of these are related to #522.  The final two commits address other problems that cropped up later in the build.

File: lib/crypto/crypto_internal.h [hunk 1]
```diff
@@ -3,7 +3,6 @@
 
 #include <stdint.h>
 
-#include <openssl/aes.h>
```
Comment (cperciva) [inline lib/crypto/crypto_internal.h:6]: This header file existed long before tarsnap 1.0.12; that's just when it was moved to its current location.

In tarsnap 0.1.5 it contained

```
struct crypto_aes_key {
        AES_KEY key;                    /* Expanded key. */
        uint64_t nonce;                 /* Nonce, used only for encryption. */
        uint8_t key_encrypted[256];     /* AES key encrypted with encr_pub. */
};
```

which is why it needed `<openssl/aes.h>`.

Can you change the commit message to just say that it was needed in previous versions but is no longer necessary?

**Comment by cperciva** [comment]
Yes, please squash the `RSA *` -> `void *` changes together.

**Comment by gperciva** [comment]
Rebased, ready for merge.

**Pull request merged**